src/Sylius/Bundle/CoreBundle/Templating/Helper/PriceHelper.php

```diff
@@ -29,11 +29,20 @@ class PriceHelper extends Helper
     private $productVariantPriceCalculator;
 
     /**
+     * @var ProductVariantPriceCalculatorInterface
+     */
+    private $productVariantOriginalPriceCalculator;
+
+    /**
      * @param ProductVariantPriceCalculatorInterface $productVariantPriceCalculator
+     * @param ProductVariantPriceCalculatorInterface $productVariantOriginalPriceCalculator
      */
-    public function __construct(ProductVariantPriceCalculatorInterface $productVariantPriceCalculator)
-    {
+    public function __construct(
+        ProductVariantPriceCalculatorInterface $productVariantPriceCalculator,
+        ProductVariantPriceCalculatorInterface $productVariantOriginalPriceCalculator
```

Comment:
I would create a new interface to split the logic

Reply:
Yes it would be better since original price is nullble and price is not. So they should not share the same calculator interface .

Reply:
why not set the original price equal to selling price and respect the same interface?

Reply:
I'm agree that null values are no more than a door for bugs to come from. And want to avoid them as possible. In best practice a method should not has multi return type. 
It's possible to update the way how a product get persisted to assure original price has an initial value(equal price). Then there is the problem of data migration to deal with also.(in order to update current products)
This is big decisions should be validated with sylius maintainers I think .

Reply:
I'm agree that null values are no more than a door for bugs to come from. And want to avoid them as possible. In best practice a method should not has multi return type. 
It's possible to update the way how a product get persisted to assure original price has an initial value(equal price). Then there is the problem of data migration to deal with also.(in order to update current products)
This is big decisions should be validated with sylius maintainers I think .

Reply:
Even though we decide to implement this feature, still `$originalPrice` and `$price` are slightly different concepts and I think it should be useful to have different interfaces for their calculators.